geoips/geoips_utils.py

```diff
@@ -445,3 +449,129 @@ def merge_nested_dicts(dest, src, in_place=True):
         raise
     if not in_place:
         return final_dest
+
+
+def expose_geoips_commands(pkg_name=None, test_log=None):
+    """Expose a list of commands that operate in the GeoIPS environment.
+
+    Where, these commands are defined under 'pyproject.toml:[tool.poetry.scripts]',
+    or 'pyproject.toml:[project.entry-points.console_scripts]'
+
+    Parameters
+    ----------
+    pkg_name: str (default = None)
+        - The name of the GeoIPS Plugin Package whose command's will be exposed.
+        - If None, assume this was called via the commandline and retrieve package_name
+          via that manner. Otherwise use the supplied package_name.
+    test_log: logging.Logger (default = None)
+        - If provided, use this log so we can test the output of this function for
+          testing purposes
+    """
+    pkg_name, log = _get_pkg_name_and_logger(pkg_name, test_log)
+    try:
+        toml_path = resources.files(pkg_name) / "../pyproject.toml"
```

Comment:
This is what I used in basically the entire CLI PRs and for ``create_plugin_registries``. If either you, @jsolbrig or @mindyls know of any other methods to access the top-folder of a plugin package other than this method, please let me know so I can use it from now on.

Reply:
I think that, if we could figure out how to do this via entrypoints rather than reading `pyproject.toml`, I would prefer to do that. It would make it independent of the build system that we're using. It would work for Poetry, Setuptools, and presumably any other build system.

Essentially, all of these scripts should be available by inspecting the entry points somehow.

Reply:
I found a way to do this. Much easier, we'll still keep the output functionality via tabulate, but this removes the need to have toml as a dependency and will work for each package.
```python
In [34]: eps = md.entry_points()

In [35]: g_scripts = list(filter(lambda ep: 'geoips' in ep.value, eps.select(group='console_scripts')))

In [36]: g_scripts
Out[36]: 
[EntryPoint(name='convert_trackfile_to_yaml', value='geoips.commandline.convert_trackfile_to_yaml:main', group='console_scripts'),
 EntryPoint(name='create_plugin_registries', value='geoips.create_plugin_registries:main', group='console_scripts'),
 EntryPoint(name='create_sector_image', value='geoips.commandline.create_sector_image:main', group='console_scripts'),
 EntryPoint(name='geoips', value='geoips.commandline.commandline_interface:main', group='console_scripts'),
 EntryPoint(name='list_available_plugins', value='geoips.commandline.list_available_plugins:main', group='console_scripts'),
 EntryPoint(name='run_procflow', value='geoips.commandline.run_procflow:main', group='console_scripts'),
 EntryPoint(name='update_tc_tracks_database', value='geoips.commandline.update_tc_tracks_database:main', group='console_scripts'),
 EntryPoint(name='xml_to_yaml_sector', value='geoips.commandline.xml_to_yaml_sector:main', group='console_scripts')]
```

Reply:
This has since been added. Let me know if you're good with it and I'll resolve this comment.

Reply:
@jsolbrig said he's good with this. We'll eventually merge this command in the CLI and call it via ``geoips list console_scripts``once the CLI has been merged in.